dipy/denoise/tests/test_noise_estimate.py

```diff
@@ -49,53 +54,68 @@ def test_piesno():
     assert_(np.abs(sigma - 50) / sigma < 0.03)
 
     sigma = _piesno_3D(rician_noise, N=1, alpha=0.01, l=1, eps=1e-10,
-                         return_mask=False,
-                         initial_estimation=initial_estimation)
+                       return_mask=False,
+                       initial_estimation=initial_estimation)
     assert_(np.abs(sigma - 50) / sigma < 0.03)
 
-    sigma = _piesno_3D(np.zeros_like(rician_noise), N=1, alpha=0.01, l=1, eps=1e-10,
-                     return_mask=False,
-                     initial_estimation=initial_estimation)
+    sigma = _piesno_3D(np.zeros_like(rician_noise), N=1, alpha=0.01, l=1,
+                       eps=1e-10, return_mask=False,
+                       initial_estimation=initial_estimation)
 
     assert_(np.all(sigma == 0))
 
-    sigma, mask = _piesno_3D(np.zeros_like(rician_noise), N=1, alpha=0.01, l=1, eps=1e-10,
-                 return_mask=True,
-                 initial_estimation=initial_estimation)
+    sigma, mask = _piesno_3D(np.zeros_like(rician_noise), N=1, alpha=0.01, l=1,
+                             eps=1e-10, return_mask=True,
+                             initial_estimation=initial_estimation)
 
     assert_(np.all(sigma == 0))
     assert_(np.all(mask == 0))
 
     # Check if no noise points found in array it exits
-    sigma = _piesno_3D(1000*np.ones_like(rician_noise), N=1, alpha=0.01, l=1, eps=1e-10,
-             return_mask=False, initial_estimation=10)
+    sigma = _piesno_3D(1000*np.ones_like(rician_noise), N=1, alpha=0.01, l=1,
+                       eps=1e-10, return_mask=False, initial_estimation=10)
     assert_(np.all(sigma == 10))
 
+
 def test_estimate_sigma():
 
     sigma = estimate_sigma(np.ones((7, 7, 7)), disable_background_masking=True)
     assert_equal(sigma, 0.)
 
-    sigma = estimate_sigma(np.ones((7, 7, 7, 3)), disable_background_masking=True)
+    sigma = estimate_sigma(np.ones((7, 7, 7, 3)),
+                           disable_background_masking=True)
     assert_equal(sigma, np.array([0., 0., 0.]))
 
-    sigma = estimate_sigma(5 * np.ones((7, 7, 7)), disable_background_masking=False)
+    sigma = estimate_sigma(5 * np.ones((7, 7, 7)),
+                           disable_background_masking=False)
     assert_equal(sigma, 0.)
 
-    sigma = estimate_sigma(5 * np.ones((7, 7, 7, 3)), disable_background_masking=False)
+    sigma = estimate_sigma(5 * np.ones((7, 7, 7, 3)),
+                           disable_background_masking=False)
     assert_equal(sigma, np.array([0., 0., 0.]))
 
     arr = np.zeros((3, 3, 3))
     arr[0, 0, 0] = 1
     sigma = estimate_sigma(arr, disable_background_masking=False, N=1)
-    assert_array_almost_equal(sigma, 0.10286889997472792 / np.sqrt(0.42920367320510366))
+<<<<<<< HEAD
```

Comment:
We use nose for testing: https://nose.readthedocs.org/en/latest/

For a gentle introduction to this approach, take a look here:
http://katyhuff.github.io/python-testing/

There are videos with a tutorial starting here:

https://www.youtube.com/watch?v=T0BE9ApIegc


On Tue, Mar 1, 2016 at 1:41 PM, Manu Tej Sharma <notifications@github.com>
wrote:

> In dipy/denoise/tests/test_noise_estimate.py
> <https://github.com/nipy/dipy/pull/957#discussion_r54640501>:
>
> >      assert_equal(sigma, np.array([0., 0., 0.]))
> >
> >      arr = np.zeros((3, 3, 3))
> >      arr[0, 0, 0] = 1
> >      sigma = estimate_sigma(arr, disable_background_masking=False, N=1)
> > -    assert_array_almost_equal(sigma, 0.10286889997472792 / np.sqrt(0.42920367320510366))
> > +<<<<<<< HEAD
>
> I would like to know more about running test on local machine.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/nipy/dipy/pull/957/files#r54640501>.
>